### PR TITLE
gogio: [MacOS] add MacOS .app compilation

### PR DESCRIPTION
**Pull request by inkeliz**
This patch is a initial implementation to make
`.app` file. It supports custom icons and sign.

Signed-off-by: Inkeliz <inkeliz@inkeliz.com>

### CODE REVIEW

**Comment by inkeliz** [comment]
Maybe, in the future, it can generate `.dmg`, which seems more common on MacOS. It can work similar to `aab/apk` and `app/ipa`.

File: gogio/macosbuild.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	if *destPath != "" {
+		if filepath.Ext(*destPath) != ".app" {
+			return fmt.Errorf("invalid output name %q, it must end with `.app`", *destPath)
+		}
+		name = filepath.Base(*destPath)
+	}
+	name = strings.TrimSuffix(name, ".app")
+
+	if bi.appID == "" {
+		return errors.New("app id is empty; use -appid to set it")
```
Comment (eliasnaur) [inline gogio/macosbuild.go:30]: Is this too strict? Isn't unsigned .app bundles runnable and therefore useful?

**Comment by eliasnaur** [comment]
Nice, thanks. Merged.

**Comment by gedw99** [comment]
> Maybe, in the future, it can generate `.dmg`, which seems more common on MacOS. It can work similar to `aab/apk` and `app/ipa`.

you can use the fyne packager to package gio apps.
just needs slight modification. i have been meaning to submit this but doubted there was any interest ?

https://github.com/fyne-io/fyne/tree/master/cmd/fyne/internal/commands

**Pull request closed without merging**